docs/DeveloperGuide.md

```diff
@@ -1,5 +1,25 @@
 # Developer Guide
 
+## Design
+### Architecture
+Given below is a quick overview of the main components of Mod Happy and how they interact with one another.
+![Class Diagram](http://www.plantuml.com/plantuml/proxy?src=https://raw.githubusercontent.com/Ch40gRv1-Mu/tp/branch-A-UML/docs/Components.puml)
+
+
```

Comment:
Seems like this section was duplicated during the merge (see lines 28-34 of the same file)